### ISSUE TITLE
Solve with relaxation=true re-initializes Gurobi solver environment

### ISSUE THREAD

**Issue by vtjeng** [JuliaOpt/JuMP.jl#1208]
```julia
using JuMP
using Gurobi

m = Model(solver = GurobiSolver(OutputFlag=0))
@variable(m, 0 <= x[i=1:10] <= i)

function upperbound_mip(x)
    @objective(x.m, Max, x)
    solve(x.m)
    return getobjectivevalue(x.m)
end

function upperbound_lp(x)
    @objective(x.m, Max, x)
    solve(x.m, relaxation=true)
    return getobjectivevalue(x.m)
end
```

```julia
>>> map(upperbound_mip, x)

Academic license - for non-commercial use only
10-element Array{Float64,1}:
  1.0
  2.0
  3.0
  4.0
  5.0
  6.0
  7.0
  8.0
  9.0
 10.0

>>> map(upperbound_lp, x)

Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
Academic license - for non-commercial use only
10-element Array{Float64,1}:
  1.0
  2.0
  3.0
  4.0
  5.0
  6.0
  7.0
  8.0
  9.0
 10.0
```

When calling `upperbound_mip` (with `relaxation=false`), we create a single Gurobi environment which is reused (as reflected in the single `Academic license message`); on ther other hand, it seems like we create multiple Gurobi environments. This makes solves with `relaxation=true` take more time than necessary (and means that aggregate solve times can be slower with `relaxation=true` if individual solves are very quick!)

This is with `julia 0.6.2`, `Gurobi.jl 0.3.3`, `JuMP 0.18.0`, and `Gurobi 7.5.2`. I did not previously observe the issue with `Gurobi 7.0.x`, but cannot confirm that it did not exist.

(I don't have any binary variables in this toy model, but I do have them in the model that I'm actually trying to solve).

**Comment by odow** [comment]
See this section in the README: https://github.com/JuliaOpt/Gurobi.jl#reusing-the-same-gurobi-environment-for-multiple-solves

```julia
julia> using JuMP, Gurobi

julia> env = Gurobi.Env()
Academic license - for non-commercial use only
Gurobi.Env(Ptr{Void} @0x0000000000193ca0)

julia> m = Model(solver=GurobiSolver(env, OutputFlag=0))
Feasibility problem with:
 * 0 linear constraints
 * 0 variables
Solver is Gurobi

julia> @variable(m, 0 <= x[i=1:10] <= i)
10-element Array{JuMP.Variable,1}:
 x[1]
 x[2]
 x[3]
 x[4]
 x[5]
 x[6]
 x[7]
 x[8]
 x[9]
 x[10]

julia> function upperbound_mip(x)
           @objective(x.m, Max, x)
           solve(x.m)
           return getobjectivevalue(x.m)
       end
upperbound_mip (generic function with 1 method)

julia> function upperbound_lp(x)
           @objective(x.m, Max, x)
           solve(x.m, relaxation=true)
           return getobjectivevalue(x.m)
       end
upperbound_lp (generic function with 1 method)

julia> map(upperbound_mip, x)
10-element Array{Float64,1}:
  1.0
  2.0
  3.0
  4.0
  5.0
  6.0
  7.0
  8.0
  9.0
 10.0

julia> map(upperbound_lp, x)
10-element Array{Float64,1}:
  1.0
  2.0
  3.0
  4.0
  5.0
  6.0
  7.0
  8.0
  9.0
 10.0
```

**Comment by vtjeng** [comment]
@odow I have been usign the solution you've proposed in my code, but it confuses me why the behavior is different when `relaxation=true` and `relaxation=false` - I would have expected a new environment to be re-created in both situations.

**Comment by odow** [comment]
If you solve the relaxation, the internal model is discarded after the solve.

https://github.com/JuliaOpt/JuMP.jl/blob/ec30d38cc95b6f114ea136972920000f03834657/src/solvers.jl#L288

I'm not sure why it is necessary. Perhaps @joehuchette can remember since [he added](https://github.com/JuliaOpt/JuMP.jl/commit/285cc24f1d08e87f4e63d1833ecad6a6201e8d57) the feature.

**Comment by joehuchette** [comment]
I don't see why it's necessary either; I'd be fine removing it. I'm guessing it was put there to deal with solvers like CPLEX that use different model types for different problem classes (LP vs IP).

**Comment by joehuchette** [comment]
Closed by #1209
